### ISSUE TITLE
[TODO] Speed Up Training

### ISSUE THREAD

**Issue by zsdonghao** [tensorlayer/hyperpose#13]
- [ ] `keypoint_random_resize` can be integrated into `keypoint_random_resize_shortestedge` with a new args `zoom_range2`.
- [ ] Data augmentation: use TensorFlow API
    - [ ] tf_keypoint_random_crop
    - [ ] tf_keypoint_random_rotate
    - [ ] tf_keypoint_random_flip
    - [ ] tf_keypoint_random_resize
    - [ ] tf_keypoint_random_resize_shortestedge
Reference: https://github.com/tensorflow/models/tree/master/research/object_detection
- [ ] Data augmentation: or Affine transform https://github.com/anatolix/keras_Realtime_Multi-Person_Pose_Estimation
- [ ] Use only 1 mask, and use TF expand and tile instead of numpy to create mask with multiple channels.
- [ ] Better prefetch setting : https://github.com/ildoonet/tf-pose-estimation/blob/master/tf_pose/pose_dataset.py#L354   https://github.com/ildoonet/tf-pose-estimation/issues/303

**Comment by zsdonghao** [comment]
Use affine transformation to replace the **for loop** coordinate transformation
- existing implementation: https://github.com/tensorlayer/tensorlayer/blob/master/tensorlayer/prepro.py#L3515
- affine transform code: https://github.com/tensorlayer/tensorlayer/blob/master/tensorlayer/prepro.py#L266
- affine transform: https://github.com/tensorlayer/tensorlayer/blob/master/tensorlayer/prepro.py#L266

**Comment by drt040312** [comment]
Hello, I am using the engineering training vgg model, but the GPU is not being used. Where do I need to perform the setup?

**Comment by drt040312** [comment]
@zsdonghao

**Comment by drt040312** [comment]
I lifted the train.py
"
# os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
# os.environ["CUDA_VISIBLE_DEVICES"] = "0"
"
Comment

Training in single simgle mode
run
Python3 train.py

Then use nvidia-smi to view the gpu usage, and did not see the training related gpu usage information.
